spec/nandi/lockfile_spec.rb

```diff
@@ -116,5 +116,52 @@
 
       persist!
     end
+
+    context "with multiple keys, not sorted by their SHA-256 hash" do
+      let(:expected_yaml) do
+        <<~YAML
+          ---
+          lower_hash:
+            foo: 5
+          higher_hash:
+            foo: 5
+        YAML
+      end
+
+      before do
+        described_class.lockfile = {
+          higher_hash: {
+            foo: 5,
+          },
+          lower_hash: {
+            foo: 5,
+          },
+        }
+      end
+
+      # This is a somewhat ridiculous trick to avoid merge conflicts in git.
+      #
+      # Normally, new migrations are added to the bottom of the Nandi lockfile.
+      # This is relatively unfriendly to git's merge algorithm, and means that
+      # if someone merges a pull request with a completely unrelated migration,
+      # you'll have to rebase to get yours merged as the last line of the file
+      # will be seen as a conflict (both branches added content there).
+      #
+      # This is in contrast to something like Gemfile.lock, where changes tend
+      # to be distributed throughout the file. The idea behind sorting by
+      # SHA-256 hash is to distribute new Nandi lockfile entries evenly, but
+      # also stably through the file. It needs to be stable or we'd have even
+      # worse merge conflict problems (e.g. if we randomised the order on
+      # writing the file, the whole thing would conflict pretty much every time
+      # it was regenerated).
```

Comment:
Instinctively I'd want this to go above the implementation rather than the test case ...

Reply:
also go to bed 💤

Reply:
On reflection, agreed. I've moved it.

Don't worry, I'm off to do my teeth! This was the result of a quick 30 minutes hacking before bedtime.